include/leveldb/iterator.h

```diff
@@ -64,6 +65,10 @@ class Iterator {
   // REQUIRES: !AtEnd() && !AtStart()
   virtual Slice value() const = 0;
 
+  // Riak specific:  if a database iterator, returns key meta data
+  // REQUIRES: Valid()
+  virtual KeyMetaData & keymetadata() const {return(keymetadata_); };
```

Comment:
keymetadata() is a const method, so it should probably return a const reference (unless there's some expected usage that requires non-const).